src/android/TP6m.jsx

```diff
@@ -106,16 +109,19 @@ class TP6M extends React.Component {
 
   async componentDidUpdate(prevProps) {
     if (
-      this.props.test === prevProps.test &&
-      this.props.platform === prevProps.platform
-    )
+      ['test', 'platform', 'past', 'ending'].every(
+        v => this.props[v] === prevProps[v]
+      )
+    ) {
       return;
+    }
 
     this.componentDidMount();
```

Comment:
`componentDidUpdate` and `componentDidMount` are long running `async` functions: They setup a promise chain, and return immediately.  They can be safely called at any time to update the component state.

I could create a third function, `async getDataAndUpdateComponent()`, and change `componentDidMount` to be  

```javascript
async componentDidMount() {
    return getDataAndUpdateComponent();
}
```

and change `componentDidUpdate` to use `getDataAndUpdateComponent()` too.  But, adding `getDataAndUpdateComponent` requires an extra named function for no benefit.

Reply:
I think the benefit is that it will be more readable for React users. It should be okay to do the alternative you are suggesting.

From https://reactjs.org/docs/react-component.html#componentdidmount

> componentDidMount() is invoked immediately after a component is mounted (inserted into the tree).

Calling this lifecycle method again will confuse users reading the docs because React users will think that this method is only called once per initialization whereas here we are calling it multiple times.